### PR TITLE
Add new endpoint to fetch "all" course logs

### LINKED ISSUE #50: Request for new API: get watch time data joined by users and media
To get the data used for visualization, now frontend has to send multiple requests through the APIs for `logs`, and usually, it costs about 30 sec or more for some popular courses like CS241.

So maybe it's time to have special APIs for each data visualization `api/.../{offeringId}`. According to the issues [#150](https://github.com/classtranscribe/FrontEnd/issues/150) and [#131](https://github.com/classtranscribe/FrontEnd/issues/131) in `FrontEnd` (Especially for [#150](https://github.com/classtranscribe/FrontEnd/issues/150), we need to have a giant table joined by users and medias, and each cell in the table represents how much time a user spent on this media). 
Hence, we need an API that can return the following data:
- An array of users who have watched videos of this offering: each object should contain (1) the user's info, requiring their `id` and `email`; (2) an array of **ALL** the medias in this offering (not just the medias the user watched). should be ordered by their `indices`. 
- Each media object should have the count of `timeupdate` event type in `last1Hr`, `last3Days`, `lastWeek`, `lastMonth`, and `total` for this user. (if a media is not watched by the user, all the values should be 0)

```json
[
  {
    "user": {
      "id" : "",
      "email": ""
    },
    "medias": [
      {
        "mediaName": "",
        "id": "",
        "last1Hr": 12,
        "last3Days": 12,
        "lastWeek": 12,
        "lastMonth": 12,
        "total": 12
      }
    ]
  }
]
```

### PR DESCRIPTION
**Pull request by asherdale**
Fixes #50 

Made a small stylistic change in a bunch of controllers just to reduce unnecessary nested if/else statements.

### CODE REVIEW

File: ClassTranscribeServer/Controllers/LogsController.cs [hunk 1]
```diff
@@ -316,10 +384,12 @@ public class StudentLog
         public class MediaLog
         {
             public string MediaId { get; set; }
+            public string MediaName { get; set; }
```
Comment (angrave) [inline ClassTranscribeServer/Controllers/LogsController.cs:377]: Add Duration too?

Comment (asherdale) [reply ClassTranscribeServer/Controllers/LogsController.cs:377]: @ruihuasui do you think we should add duration? Would it be helpful to have?

Comment (ruihuasui) [reply ClassTranscribeServer/Controllers/LogsController.cs:377]: Yeah, It sounds good to have a duration.

**Review by angrave — APPROVED** [review]

**Comment by angrave** [comment]
Please delete branch if you're done with it

**Pull request merged**